src/android/unregisterNativeModule.js

```diff
@@ -4,30 +4,32 @@ const path = require('path');
 const compose = require('lodash').flowRight;
 const getReactVersion = require('../getReactNativeVersion');
 const getPrefix = require('./getPrefix');
+const makeSettingsPatch = require('./patches/makeSettingsPatch');
+const makeBuildPatch = require('./patches/makeBuildPatch');
+
+const isInstalled = require('./isInstalled');
 
 const cut = (scope, pattern) =>
   scope.replace(pattern, '');
 
 module.exports = function unregisterNativeAndroidModule(name, dependencyConfig, projectConfig) {
   const prefix = getPrefix(getReactVersion(projectConfig.folder));
 
+  const settingsPatch = makeSettingsPatch(name, dependencyConfig, {}, projectConfig).patch;
+  const buildPatch = makeBuildPatch(name).patch;
+
   /**
    * @param  {String} content Content of the Settings.gradle file
    * @return {String}         Patched content of Settings.gradle
    */
-  const cutModuleFromSettings = (name) => (content) =>
-    cut(content, `include ':${name}'\n` +
-      `project(':${name}').projectDir = ` +
-      `new File(rootProject.projectDir, '../node_modules/${name}/android')\n`
-    );
+  const cutModuleFromSettings = (content) => cut(content, settingsPatch);
```

Comment:
So the issue for settings patch was - sometimes path is android/app and that will fail to discover it. Decided to just get a settings patch and search for it which is more future proof if we add more things in the future.